src/common/qat.c

```diff
@@ -0,0 +1,537 @@
+
+#define PHYS_CONTIG_FREE(pMemAddr) Mem_Free_Contig((void *)&pMemAddr)
+
+extern CpaStatus qaeMemInit(void);
+extern void qaeMemDestroy(void);
+
+/* Threads */
+typedef pthread_t sampleThread;
+static sampleThread gPollingThread;
+static volatile int gPollingCy = 0;
+
+static __inline CpaStatus Mem_OsMemAlloc(void **ppMemAddr, Cpa32U sizeBytes)
+{
+    *ppMemAddr = malloc(sizeBytes);
+    if (NULL == *ppMemAddr)
+    {
+        return CPA_STATUS_RESOURCE;
+    }
+    return CPA_STATUS_SUCCESS;
+}
+
+static __inline CpaStatus Mem_Alloc_Contig(void **ppMemAddr,
+                                           Cpa32U sizeBytes,
+                                           Cpa32U alignment)
+{
+    *ppMemAddr = qaeMemAllocNUMA(sizeBytes, 0, alignment);
+    if (NULL == *ppMemAddr)
+    {
+        return CPA_STATUS_RESOURCE;
+    }
+    return CPA_STATUS_SUCCESS;
+}
+
+static __inline void Mem_OsMemFree(void **ppMemAddr)
+{
+    if (NULL != *ppMemAddr)
+    {
+        free(*ppMemAddr);
+        *ppMemAddr = NULL;
+    }
+}
+
+static __inline void Mem_Free_Contig(void **ppMemAddr)
+{
+    if (NULL != *ppMemAddr)
+    {
+        qaeMemFreeNUMA(ppMemAddr);
+        *ppMemAddr = NULL;
+    }
+}
+
+static __inline CpaStatus OS_SLEEP(Cpa32U ms)
+{
+    int ret = 0;
+    struct timespec resTime, remTime;
+    resTime.tv_sec = ms / 1000;
+    resTime.tv_nsec = (ms % 1000) * 1000000;
+    do
+    {
+        ret = nanosleep(&resTime, &remTime);
+        resTime = remTime;
+    } while ((ret != 0) && (errno == EINTR));
+
+    if (ret != 0)
+    {
+        return CPA_STATUS_FAIL;
+    }
+    else
+    {
+        return CPA_STATUS_SUCCESS;
+    }
+}
+
+static __inline CpaPhysicalAddr virtToPhys(void *virtAddr)
+{
+    return (CpaPhysicalAddr)qaeVirtToPhysNUMA(virtAddr);
+}
+
+static __inline CpaStatus threadCreate(sampleThread *thread,
+                                             void *funct,
+                                             void *args)
+{
+    if (pthread_create(thread, NULL, funct, args) != 0)
+    {
+        printf("Failed create thread\n");
+        return CPA_STATUS_FAIL;
+    }
+    else
+    {
+        pthread_detach(*thread);
+        return CPA_STATUS_SUCCESS;
+    }
+}
+
+static __inline void threadExit(void)
+{
+    pthread_exit(NULL);
+}
+
+static void getInstance(CpaInstanceHandle *pCyInstHandle)
+{
+    CpaInstanceHandle cyInstHandles[MAX_INSTANCES];
+    Cpa16U numInstances = 0;
+    CpaStatus status = CPA_STATUS_SUCCESS;
+
+    *pCyInstHandle = NULL;
+    status = cpaCyGetNumInstances(&numInstances);
+
+    if (numInstances >= MAX_INSTANCES)
+    {
+        numInstances = MAX_INSTANCES;
+    }
+    if ((status == CPA_STATUS_SUCCESS) && (numInstances > 0))
+    {
+        status = cpaCyGetInstances(numInstances, cyInstHandles);
+        if (status == CPA_STATUS_SUCCESS)
+            *pCyInstHandle = cyInstHandles[0];
+    }
+
+    if (0 == numInstances)
+    {
+        printf("No QAT instances found.\n");
+    }

```

Comment:
Would be great to follow our coding style in this file too.
e.g.
static void
getInstance(CpaInstanceHandle *pCyInstHandle)
{
    ...
    if (numInstances >= MAX_INSTANCES) {
            numInstances = MAX_INSTANCES;
    }

Reply:
also, don't need curly brackets for single line body of if statement

Reply:
Fixed.